icp/src/python/triton_distributed/icp/eventplane_nats.py

```diff
@@ -0,0 +1,60 @@
+import uuid
+from datetime import datetime
+from typing import Optional
+
+import nats
+
+from triton_distributed.icp.eventplane import Event, EventTopic
+from triton_distributed.icp.protos import event_pb2
+
+
+class EventPlaneNats:
+    """EventPlane implementation using NATS."""
+
+    def __init__(self, server_url: str, component_id: uuid.UUID):
+        self.server_url = server_url
```

Comment:
```suggestion
        self._server_url = server_url
```
Would make these private or we can add to abstract class as abstract property.